File: 941-sort-array-by-parity/sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int i=-1,j=0;
        List<Integer> numsl = Arrays.stream(nums).boxed().collect(Collectors.toList());
        for(;j<numsl.size();j++)
        {
            if(numsl.get(j)%2==0)
            {
                Collections.swap(numsl,++i,j);
            }
        }
        return numsl.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 941-sort-array-by-parity/sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int i=-1,j=0;
        for(;j<nums.size();j++)
        {
            if(nums[j]%2==0)
            {
                i++;
                int tmp = nums[i];
                nums[i]=nums[j];
                nums[j]=tmp;
            }
        }
        return nums;
    }
};